airflow/task/task_runner/standard_task_runner.py

```diff
@@ -186,3 +193,20 @@ def get_process_pid(self) -> int:
         if self.process is None:
             raise RuntimeError("Process is not started yet")
         return self.process.pid
+
+    def _read_task_utilization(self):
+        dag_id = self._task_instance.dag_id
+        task_id = self._task_instance.task_id
+
+        while True:
+            try:
+                with self.process.oneshot():
+                    mem_usage = self.process.memory_percent()
+                    cpu_usage = self.process.cpu_percent(interval=1)
```

Comment:
What I found that `interval` do not work well with `oneshot()` and always return `0`

However if we do not provide interval (`None` by default) and manually sleep then it works as expected.

Reply:
Interesting! Thanks for the info!